taskomatic/jobs/stale_systems.py

```diff
@@ -11,32 +11,54 @@
 
 def run():
     """Application entrypoint"""
-    LOGGER.info('Started stale_systems job.')
+    LOGGER.info("Started stale_systems job.")
 
     conn = get_conn()
     cur = conn.cursor()
-    updated = 0
-
-    while True:
-        cur.execute("""SELECT id from system_platform sp
-                    WHERE when_deleted IS NULL
-                      AND stale_warning_timestamp IS NOT NULL
-                      AND stale = 'F'
-                      AND now() > stale_warning_timestamp LIMIT 1 FOR UPDATE OF sp""")
-        system_id = cur.fetchone()
-        if not system_id:
-            break
-        cur.execute("""UPDATE system_platform SET stale = 'T' where id = %s""", system_id)
-        conn.commit()
-        updated += 1
+
+    query = """
+        UPDATE system_platform
+        SET stale = TRUE
+        FROM (
+            SELECT id
+            FROM system_platform
+            WHERE when_deleted IS NULL
+            AND stale_warning_timestamp IS NOT NULL
+            AND stale = FALSE
+            AND now() > stale_warning_timestamp
+            ORDER BY id
+        ) AS subquery
+        WHERE system_platform.id = subquery.id
+    """
+    cur.execute(query)
+    LOGGER.info("Marked %s systems as stale", cur.rowcount)
+    conn.commit()
     cur.close()
-    conn.close()
 
-    LOGGER.info('Marked %s systems as stale.', updated)
+    cur = conn.cursor()
+    query = """
+        UPDATE system_platform
+        SET when_deleted = now()
+        FROM (
+            SELECT id
+            FROM system_platform
+            WHERE when_deleted IS NULL
+            AND culled_timestamp IS NOT NULL
+            AND stale = TRUE
+            AND now() + INTERVAL '1 day' > culled_timestamp
```

Comment:
I think it should be rather:
`now() > culled_timestamp + INTERVAL '1 day'`

to delete such systems one day later

Reply:
fixed